lib/cext/preprocess.rb

```diff
@@ -131,20 +153,268 @@ def write_array(name, leaking)
 end
 
 def read_write_array(name, leaking)
-  [read_array(name), 
+  [read_array(name),
    write_array(name, leaking)]
 end
 
+def cast_value_for_native(name, suffix = '', type = :local)
```

Comment:
What does this function solve? Could you add a comment (same for other functions)?